app/controllers/dashboard/documents_controller.rb

```diff
@@ -29,8 +29,8 @@ class Dashboard::DocumentsController < Dashboard::BaseController
 
   def index
     @documents          = @protocol.documents
-    @permission_to_edit = @protocol.project_roles.where(identity: @user, project_rights: ['approve', 'request']).any?
-    permission_to_view  = @protocol.project_roles.where(identity: @user, project_rights: ['approve', 'request', 'view']).any?
+    @permission_to_edit = @protocol.permission_to_edit?(@user)
```

Comment:
Moved this logic to the protocol level so we're not constantly duplicating the code